compiler/src/iree/compiler/Codegen/Common/ReconcileTranslationInfo.cpp

```diff
@@ -61,18 +61,19 @@ static FailureOr<int64_t> reconcileSubgroupSize(
   return subgroupSize;
 }
 
-/// Helper function to retrieve the waves-per-eu value from translation info.
-static std::optional<int64_t>
-getWavesPerEu(IREE::Codegen::TranslationInfoAttr translationInfo) {
+/// Helper function to retrieve the target-func-attrs value from translation
+/// info.
+static DictionaryAttr
+getTargetFuncAttrs(IREE::Codegen::TranslationInfoAttr translationInfo) {
   auto translationConfig = translationInfo.getConfiguration();
   if (!translationConfig) {
-    return std::nullopt;
+    return nullptr;
   }
-  auto attr = translationConfig.getAs<IntegerAttr>("waves_per_eu");
+  auto attr = translationConfig.getAs<DictionaryAttr>("target_func_attrs");
```

Comment:
should we call this `llvm_func_attrs`, since it only applies to llvm compilations?

Reply:
spirv decorations are more structures, I'm not sure if we can make it work the same way. I'd rather keep it contained to llvm and expand later if we need to.

Reply:
Ok fair, let me change it then